gwcs/wcs.py

```diff
@@ -1432,11 +1426,12 @@ def fix_inputs(self, fixed):
 
     def to_fits_sip(self, bounding_box=None, max_pix_error=0.25, degree=None,
                     max_inv_pix_error=0.25, inv_degree=None,
-                    npoints=32, crpix=None, verbose=False):
+                    npoints=32, crpix=None, projection='TAN',
```

Comment:
The code will not work correctly with arbitrary projections yet. Non zenithal projections or projections which require parameters are currently not supported. The safest choice is to use `TAN` until the other limitations are fixed.

Reply:
Added support for arbitrary projections by allowing users to supply a fully instantiated object with whatever parameters they need.